transcom/mymove
# INT B-18824 requested office users get single user endpoint

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3a907137)

## Summary

***please ignore my brainfart*** the branch has the incorrect BL item, it should be `B-18824` and not `B-18885`

When it comes to having office users requesting their account, the admins will eventually need to review that user in the Admin console within MilMove. This will require an endpoint that can be called to retrieve that information so the UI can display it. I took notes off of what was being done for viewing admin users and pretty much copied that functionality over to `requested_office_users`.

I have merged the branch of [B-18822](https://github.com/transcom/mymove/pull/12113) into this in order to copy over the creation of the endpoint done in that branch.

The return value of querying the db should look like this:
```
{
  "id": "c56a4180-65aa-42ec-a945-5fd21dec0538",
  "userId": "3fa85f64-5717-4562-b3fc-2c963f66afa6",
  "firstName": "string",
  "middleInitials": "string",
  "lastName": "string",
  "email": "dkj-9pUmr@uLdkqVEkf9MsW9h8yoezuK-Eg5HEirPkavMgiFw8tvPyuVMaKQy3uys3ig3IrQM5WOyg9x2yvOsc3K1M2kv.rBrA5hihPdWu3.kYiMtccDklvvtbpSBUutDabBbbDQwVkCdVbTDmiKssUrhyvBpMGdnTAgvrsTONFhcPznXnAPeydXIKvZmBFqBYobvctock",
  "telephone": "468-681-8762",
  "transportationOfficeId": "3fa85f64-5717-4562-b3fc-2c963f66afa6",
  "active": true,
  "roles": [
    {
      "id": "c56a4180-65aa-42ec-a945-5fd21dec0538",
      "roleType": "customer",
      "roleName": "Transportation Ordering Officer",
      "createdAt": "2024-03-01T14:36:30.952Z",
      "updatedAt": "2024-03-01T14:36:30.952Z"
    }
  ],
  "edipi": "string",
  "otherUniqueId": "string",
  "rejectionReason": "string",
  "status": "APPROVED",
  "createdAt": "2024-03-01T14:36:30.952Z",
  "updatedAt": "2024-03-01T14:36:30.952Z"
}
```

This PR does the following:
- adds `GET requested-office-users/{officeUserId}` endpoint that returns a single row in the `office_users` table
- updates `swagger-def` to include the endpoint and handle the return values
- updated and added relevant tests

### How to test

1. Fire up `client` & `server`
2. Access MM as an admin
3. It isn't necessary, but make sure you have some users in the `office_users` table, they don't NEED to have a `REQUESTED` status, this query should still return a user
4. Head on over to `http://adminlocal:3000/swagger-ui/admin.html#/`
5. Copy an office user's `id` from the `office_users` table
6. Find the `requested-office-users/{officeUserId}` endpoint and plug in that `id`
7. Execute that sucker and you should get a return value back

## Screenshots
![Screenshot 2024-03-01 at 8 36 41 AM](https://github.com/transcom/mymove/assets/136510600/ecd6a649-e5bc-4304-a2b0-91e565696220)
